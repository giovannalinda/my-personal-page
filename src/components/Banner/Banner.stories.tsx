import { Meta, Story } from '@storybook/react'
import { Banner } from './Banner'

export default {
  title: 'Components/Banner',
  component: Banner,
} as Meta

export const Default: Story = () => <Banner />
