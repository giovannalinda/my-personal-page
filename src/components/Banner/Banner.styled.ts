import styled from 'styled-components'
import { theme } from '~/styles'

export const Container = styled.section`
  display: flex;
  align-items: center;
  justify-content: center;
  margin-bottom: 34rem;
  text-align: center;

  @media screen and (max-width: ${theme.breakpoints.smallDesktop}) {
    padding: ${theme.space[5]};
  }
`

export const Title = styled.h1`
  font-size: ${theme.font.sizes.title};
  color: ${theme.colors.white};
  -webkit-text-fill-color: ${theme.colors.black};
  -webkit-text-stroke-width: 1px;
  -webkit-text-stroke-color: ${theme.colors.white};
`
