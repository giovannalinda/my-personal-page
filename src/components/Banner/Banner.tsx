import { motion } from 'framer-motion'
import { InView } from 'react-intersection-observer'

import * as S from './Banner.styled'

const TITLE = 'Vamos fazer algo incr√≠vel juntos.'

const letterVariant = {
  hidden: { opacity: 0 },
  visible: { opacity: 1 },
}

export function Banner() {
  return (
    <InView triggerOnce>
      {({ inView, ref }) => (
        <S.Container ref={ref}>
          {inView && (
            <S.Title>
              {TITLE.split('').map((char, index) => (
                <motion.span
                  key={`${char}-${index}`}
                  variants={letterVariant}
                  transition={{
                    delay: index / TITLE.length,
                    duration: 0.4,
                  }}
                  initial='hidden'
                  animate='visible'
                >
                  {char}
                </motion.span>
              ))}
            </S.Title>
          )}
        </S.Container>
      )}
    </InView>
  )
}
