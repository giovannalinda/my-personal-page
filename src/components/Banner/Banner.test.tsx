import { render } from '~/__helpers__/app-tests'
import { Banner } from './Banner'

describe('Banner', () => {
  it('matches snapshot', () => {
    const { container } = render(<Banner />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
