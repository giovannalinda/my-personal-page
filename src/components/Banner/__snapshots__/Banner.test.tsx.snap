// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner matches snapshot 1`] = `
<section
  class="sc-bczRLJ jFKiRm"
>
  <h1
    class="sc-gsnTZi dWprHZ"
  >
    Vamos fazer algo incr√≠vel juntos.
  </h1>
</section>
`;
